nemo/collections/asr/modules/conv_asr.py

```diff
@@ -218,16 +208,27 @@ def update_max_sequence_length(self, seq_length: int, device):
             seq_length = global_max_len.int().item()
 
         if seq_length > self.max_audio_length:
-            self.max_audio_length = seq_length * 2
+            if seq_length < 5000:
+                seq_length = seq_length * 2
+            elif seq_length < 10000:
+                seq_length = seq_length * 1.5
+            self.max_audio_length = seq_length
+
+            device = next(self.parameters()).device
+            seq_range = torch.arange(0, self.max_audio_length, device=device)
+            if hasattr(self, 'seq_range'):
+                self.seq_range = seq_range
+            else:
+                self.register_buffer('seq_range', seq_range, persistent=False)
 
             # Update all submodules
             for name, m in self.named_modules():
                 if isinstance(m, MaskedConv1d):
-                    if m.use_mask:
-                        m.update_masked_length(self.max_audio_length, device=device)
-
+                    m.max_len = self.max_audio_length
```

Comment:
Can we update the jasper method update_masked_length() to take these two values instead of access inner members like this? Or even have a separate function to update these lengths in place without allocating.

Reply:
done.